graphene_sqlalchemy/fields.py

```diff
@@ -94,18 +98,46 @@ def __init__(self, type, *args, **kwargs):
         super(SQLAlchemyConnectionField, self).__init__(type, *args, **kwargs)
 
 
-__connectionFactory = UnsortedSQLAlchemyConnectionField
+def default_connection_field_factory(relationship, registry):
+    model = relationship.mapper.entity
+    _type = registry.get_type_for_model(model)
+    return UnsortedSQLAlchemyConnectionField(_type._meta.connection)
+
+
+__connection_field_factory =  default_connection_field_factory
```

Comment:
The current pattern to register a factory is to use a global module variable. I followed it in this PR but I think we should move away from it because:
- unless every module that defines a subclass of SQLAlchemyObjectType explicitly calls `register_connection_field_factory`, you may not get the expected connection when you import a `SQLAlchemyObjectType` subclass. 
- it is not possible to use different factories within the same schema.

Instead I would prefer to explicitly define the factory on the `Meta` class of `SQLAlchemyObjectType`:
```python
class MyType(SQLAlchemyObjectType):
     class Meta:
         connection_field_factory = ...
         ...
```

This behavior could easily be shared within an app by putting this in an abstract meta class. Let me know if you agree with this change and I'll update this PR.

Reply:
Clearly it seems more "standard" to implement it as meta parameter

But think also to retro-compatibility if you change the behavior (with deprecated warnings as you wrote for `register_connection_field_factory`)
Maybe you could do it in another PR, because this is a different subject

Reply:
My worry is that if we merge this as is now we'll have to support 2 deprecated methods (`register_connection_field_factory` and `registerConnectionFieldFactory) with different interfaces. In my opinion it would easier / cleaner to fix everything at once if you also think that 'meta' is the way to go. What do you think?

Reply:
Clearly, I think the "meta" approach is the right way.

My point is just : if you implement this with the "meta" way, do you still keep a "retro-compatibility" on the old `registerConnectionFieldFactory` ? I think it's important